docs/clipboard-pickling/explainer.md

```diff
@@ -115,11 +115,33 @@ Sites can read or write pickled versions of sanitized formats, by providing the
 
-On Windows, clipboard formats are often named using capital words separated by a space. Therefore, a MIME type `"custom/format"` will be transformed to `"Web Custom Format`”. Note that the prefix `"Web"` precedes the transformed format name.
+There will be a custom format map which contains a JSON payload that will have a mapping of custom format MIME type to web custom format. There can only be 100 custom format per user session and it will be registered when the web authors request for a custom format. Since this format is allocated in the global atom space on Windows, even if the sites register it multiple times, the format strings will only be allocated once and `RegisterClipboardFormat` system function call will just return the unique value corresponding to that format.
 
-On Linux, ChromeOS, and Android, MIME types are often used (though Linux and ChromeOS don’t have strong recommendations regarding clipboard format naming conventions). Therefore, a MIME type `"custom/format"` will be transformed to `"application/web;type="custom/format""`. Note that the pickling prefix `"application/web;type="` precedes the transformed format name, which has quotes around the format name.
+Native apps will need to read the web custom format map and parse its payload as an alternative to `EnumFormats` on Windows to fetch the mapping of the custom MIME type to web custom format. Using this the native app can then request for the web custom format corresponding to a particular MIME type of interest. This also helps in [Delay Rendering](https://docs.microsoft.com/en-us/windows/win32/dataxchg/clipboard-operations#delayed-rendering) of formats on Windows where the native app just provides an indication to the clipboard that a format is available and when it is requested by some other app, it renders the content on-demand.
+
+The web custom format map will have the below naming convention per platform:
+
+On Windows it will be inserted as `Web Custom Format Map`, on MacOS `com.web.custom.format.map` & On Linux/Android/CrOS etc `application/web;type="web/customformatmap"`.
+The payload in this format map will be of type JSON with the key representing the MIME type and the web custom format as the value.
+e.g. On Windows the web custom format map will have the below payload:
+```
+{
+
+  "text/html" : "Web Custom Format0",
+  "text/plain" : "Web Custom Format1",
+  "text/csv" : "Web Custom Format2"
+}
+```
+
+The web custom format in the value contains the actual payload of the custom MIME type. It will be serialized in terms of raw bytes and will have the below formating naming convention:
+
+On MacOS (and iOS), clipboard formats are named using the UTI reverse-DNS naming convention. Therefore, a MIME type `"custom/format"` will be transformed to `"com.web.custom.format(0-99)"`. Note that the pickling prefix `"com.web"` precedes the transformed format name `"custom.format"`, and the slash is converted to a period.
```

Comment:
Where is "custom/format" coming from? I thought this will be provided by the browser, so sites and native apps both don't need to supply this

Reply:
so `custom/format` is provided by the web authors. That will be mapped to a `web custom format[0-99]`. I guess I should use a different MIME type so there is no confusion.

Reply:
Ooh yeah, it read to me like if the web author provided "foo/type", it will be transformed to `"com.web.foo.type(0-99)"`. From our discussions, it seems instead like the web-provided type won't affect naming of OS-registered formats.

Maybe we could mention the web-provided format in line 138 in the more platform-neutral explanation, then just discuss how each platform will name the container formats here?

Reply:
Sounds good, I'll make the changes.